public/app/core/components/AppChrome/AppChrome.tsx

```diff
@@ -53,6 +54,18 @@ export function AppChrome({ children }: Props) {
     chrome.setMegaMenuOpen(!state.megaMenuOpen);
   };
 
+  const shouldShowReturnToPrevious = () => {
+    if (state.returnToPrevious && location.pathname === state.returnToPrevious.href) {
+      setReturnToPrevious({ title: '', href: '' });
+    }
+    return (
+      state.returnToPrevious &&
+      state.returnToPrevious.href &&
+      state.returnToPrevious.title &&
+      location.pathname !== state.returnToPrevious.href
```

Comment:
I think this  is a really good example of why setting values to empty string to denote 'it's empty' is annoying - 

The ReturnToPrevious type defines href and title as 

```ts
interface ReturnToPreviousProps {
  href: string;
  title: string;
}
```

When I read this type, I make an assumption (for better or worse) that the href value is always there. But that's not quite true because we use empty string to say its undefined, which leads to very verbose checks like the above. 

Instead, just knowing that i have a `ReturnToPrevious` object should be enough to tell me that I must render the button.